### PR TITLE
Increase test coverage for TimezoneProvider

### PR DESCRIPTION
**Pull request by sergeyklay**
### 🎯 Scope & Context
**Type:** Chore  
**Intent:** Add comprehensive unit tests for `TimezoneProvider` to validate cookie read/write behavior, timezone offset → hour conversion, fractional-offset tolerance, single-run guards, multi-cookie parsing, and edge-case handling to ensure reliable server-side timezone detection.

### 🧭 Reviewer Guide
**Complexity:** Low

#### Entry Point
Start with `components/common/TimezoneProvider.test.tsx` — this file contains the full test suite (12 tests) covering rendering, cookie setting format (path, SameSite, max-age ≈ 1 year), conversion of minute offsets to hour values (including fractional offsets), tolerance checks, no-op when cookie matches, update when differing, single-execution behavior, multiple-cookie parsing, and invalid-value handling. Then review `components/common/TimezoneProvider.tsx` to confirm the implementation under test: useEffect with hasRun guard, calculation of `offsetHours = -(new Date().getTimezoneOffset() / 60)`, existing cookie parsing, tolerance threshold (Math.abs < 0.01), and cookie-writing string composition.

#### Sensitive Areas
- `components/common/TimezoneProvider.test.tsx` - cookie mocking (`document.cookie` getter/setter), lifecycle of mocks (`beforeEach`/`afterAll`), and Jest spies on `Date.prototype.getTimezoneOffset`.
- `components/common/TimezoneProvider.tsx` - offset sign/scale conversion, tolerance value (0.01) for fractional offsets, cookie parsing logic (splitting `document.cookie`), and the exact cookie attributes (`path=/; max-age=365*24*60*60; SameSite=Lax`).

### ⚠️ Risk Assessment
- **Breaking Changes:** No breaking changes - tests only, no production code modifications.  
- **Migrations/State:** No migrations or state changes

### CODE REVIEW

File: components/common/TimezoneProvider.test.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    expect(cookieSetter).toHaveBeenCalledWith(
+      expect.stringContaining(`${COOKIE_NAME}=1`)
+    );
+    expect(cookieSetter).toHaveBeenCalledWith(
+      expect.stringContaining('path=/')
+    );
+    expect(cookieSetter).toHaveBeenCalledWith(
+      expect.stringContaining('SameSite=Lax')
+    );
+  });
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply components/common/TimezoneProvider.test.tsx]: Should be fixed now, please recheck

File: components/common/TimezoneProvider.test.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  beforeEach(() => {
+    jest.clearAllMocks();
+    cookieGetter = jest.fn().mockReturnValue('');
+    cookieSetter = jest.fn();
+    Object.defineProperty(document, 'cookie', {
+      configurable: true,
+      get: cookieGetter,
+      set: cookieSetter,
+    });
+  });
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply components/common/TimezoneProvider.test.tsx]: Should be fixed now, please recheck
